src/libs/actions/IOU.ts

```diff
@@ -807,16 +807,18 @@ function buildOnyxDataForMoneyRequest(
     if (!policy || !PolicyUtils.isPaidGroupPolicy(policy)) {
         return [optimisticData, successData, failureData];
     }
+    const isPartialTransaction = TransactionUtils.isMerchantMissing(transaction) && TransactionUtils.isAmountMissing(transaction);
```

Comment:
Should we use method `isPartialMerchant` here? I saw @cead22 also mentioned the merchant value `(none)` means partial transaction [here](https://github.com/Expensify/App/issues/38131#issuecomment-2010830030).

Reply:
isMerchantMissing already checks for `(none)`.
https://github.com/Expensify/App/blob/69a2849f163640021e3248eda536d2db112cb153/src/libs/TransactionUtils.ts#L140-L147

Reply:
Yeah, I'm just not sure if this case ` || transaction?.merchant === ''` also means the transaction is partial.

Reply:
@cead22 can you confirm whether empty merchant is also considered partial? Or is it only `(none)`?

Reply:
Only `(none)` according to our PHP backend code, and our JS code for old dot
```php
    /**
     * Merchant name used to identify partial transactions.
     *
     * @var string
     */
    public const PARTIAL_MERCHANT_NAME = '(none)';

...

    /**
     * Is this transaction a partial transaction?
     *
     * @return bool
     */
    public function isPartial()
    {
        return $this->getMerchant() === $this::PARTIAL_MERCHANT_NAME && $this->getAmount() === $this::PARTIAL_AMOUNT;
    }
```
```js
/**
 * Returns whether the expense is a complete expense, ie,
 * the merchant name isn't '(none)' OR the amount isn't 0
 * @returns {Boolean}
 */
SExpense3.prototype.isComplete = function () {
    return this.getMerchant() !== '(none)' || this.getAmount() !== 0;
};

/**
 * Returns whether the expense is a partial expense, ie,
 * the merchant name is '(none)' AND the amount is 0
 * @returns {Boolean}
 */
SExpense3.prototype.isPartial = function () {
    return !this.isComplete();
};
```

Reply:
@cead22 Thanks!

I just realized when we create a new scan request, the merchant is actually `(none)` but we don't show it. I have updated to use `isPartialMerchant` instead.

@eh2077